modules/distributor/distributor.go

```diff
@@ -526,18 +533,24 @@ func (d *Distributor) UsageTrackerHandler() http.Handler {
 
 // requestsByTraceID takes an incoming tempodb.PushRequest and creates a set of keys for the hash ring
 // and traces to pass onto the ingesters.
-func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int) ([]uint32, []*rebatchedTrace, error) {
+func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount, maxSpanAttrSize int) ([]uint32, []*rebatchedTrace, int, error) {
 	const tracesPerBatch = 20 // p50 of internal env
 	tracesByID := make(map[uint32]*rebatchedTrace, tracesPerBatch)
+	discardedSpans := 0
 
 	for _, b := range batches {
 		spansByILS := make(map[uint32]*v1.ScopeSpans)
 
 		for _, ils := range b.ScopeSpans {
 			for _, span := range ils.Spans {
+				// drop spans with attributes that are too large
+				if spanContainsAttributeTooLarge(span, maxSpanAttrSize) {
```

Comment:
If there is a way we could drop only the attribute that seems better than dropping the span (preserve as much information as possible).  It doesn't fit as easily within our current metrics/logs though.